cirq/ops/linear_combinations.py

```diff
@@ -205,3 +207,54 @@ def extend(expansion: value.LinearDict[str],
             extended_expansion = extend(expansion, op.qubits, self.qubits)
             result += extended_expansion * coefficient
         return result
+
+
+class PauliSum:
+    """Represents operator defined by linear combination of PauliStrings.
+
+    Since PauliStrings store their own coefficients, this class
+    does not implement the LinearDict interface. Instead, you can
+    add and subtract terms and then iterate over the resulting
+    (simplified) expression.
+
+    Under the hood, this class is backed by a LinearDict with coefficient-less
+    PauliStrings as keys. PauliStrings are reconstructed on-the-fly during
+    iteration.
+
+    """
+
+    def __init__(self, linear_dict):
+        self._linear_dict = linear_dict
+
+    @classmethod
+    def from_pauli_strings(cls, terms: List[PauliString]):
+        terms = {pstring.unit(): pstring.coefficient for pstring in terms}
+        return cls(linear_dict=value.LinearDict(terms))
+
+    def __iter__(self):
+        for vec, coeff in self._linear_dict.items():
+            yield PauliString.from_unit(vec, coeff)
+
+    def __iadd__(self, other):
```

Comment:
`__add__`, `__sub__`, `__mul__`, `__rmul__`, `__imul__`, and `__truediv__` should also be defined.

truediv and rmul only need to work in the case where the other value is a number, but the others should work for number or PauliString or PauliSum.

Reply:
I had this sitting on a branch for a long time as inheriting from `LinearDict` so all of these got filled in. I got back to it today and ripped it out to favor composition and all the dundermethods were left behind in my haphazardness :)